sdk/go/common/workspace/plugins.go

```diff
@@ -190,12 +190,22 @@ func (err *MissingError) Error() string {
 // PluginSource deals with downloading a specific version of a plugin, or looking up the latest version of it.
 type PluginSource interface {
 	// Download fetches an io.ReadCloser for this plugin and also returns the size of the response (if known).
-	Download(
+	Download(version semver.Version, opSy string, arch string,
+		getHTTPResponse func(*http.Request) (io.ReadCloser, int64, error)) (io.ReadCloser, int64, error)
+	// DownloadWithContext fetches an io.ReadCloser for this plugin and also returns the size of the response
+	// (if known). This variant takes a context to facilitate cancellation.
+	DownloadWithContext(ctx context.Context,
```

Comment:
I'd rather see a breaking change here then a duplicate method on the `PluginSource` interface.

Reply:
Isn't adding `DownloadWithContext` et. al. to this interface a breaking change in itself? Doing it in a truly non-breaking way would require a new interface, right? (And then any downstream implications of that).

Is doing all that really worth it in this case? I don't have the data to back it up, but I'd be shocked if this is being used externally (aside from any forks of the CLI). This API is not used by users writing Pulumi Go programs. Can we be pragmatic here?

Reply:
No, you're right, good catch, will fix. Are there any other SxS versions you would ditch and just fix in place?